docusaurus/docs/Android/03-ui/04-message-components/05-message-composer.mdx

```diff
@@ -0,0 +1,370 @@
+
+To use `MessageComposerView`, include it in your XML layout.
+
+```xml
+<io.getstream.chat.android.ui.message.composer.MessageComposerView
+    android:id="@+id/messageComposerView"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content" />
+```
+
+The recommended way of setting up `MessageComposerView` is by binding it to the `MessageComposerViewModel`. This will make it fully functional by setting up any necessary listeners and data handling.
+
+```kotlin
+class ChatFragment : Fragment() {
+    lateinit var messageComposerView: MessageComposerView
+
+    // Create MessageComposerViewModel for a given channel
+    val factory = MessageListViewModelFactory(cid = "channelType:channelId")
+    val messageComposerViewModel: MessageComposerViewModel by viewModels { factory }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        // Bind MessageComposerViewModel with MessageComposerView
+        messageComposerViewModel.bindView(messageComposerView, viewLifecycleOwner)
+    }
+}
+```
+
+Because it doesn't make sense to use the `MessageComposerView` as a standalone component, you also need to integrate it with the `MessageListView`:
+
+```kotlin
+class ChatFragment : Fragment() {
+    lateinit var messageComposerView: MessageComposerView
+    lateinit var messageListView: MessageListView
+
+    // Create ViewModels for MessageComposerView and MessageListView
+    val factory = MessageListViewModelFactory(cid = "channelType:channelId")
+    val messageComposerViewModel: MessageComposerViewModel by viewModels { factory }
+    val messageListViewModel: MessageListViewModel by viewModels { factory }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        // Bind MessageComposerViewModel with MessageComposerView
+        messageComposerViewModel.bindView(messageComposerView, viewLifecycleOwner)
+
+        // Integrate MessageComposerView with MessageListView
+        messageListViewModel.mode.observe(viewLifecycleOwner) {
+            when (it) {
+                is MessageListViewModel.Mode.Thread -> {
+                    messageComposerViewModel.setMessageMode(MessageMode.MessageThread(it.parentMessage))
+                }
+                is MessageListViewModel.Mode.Normal -> {
+                    messageComposerViewModel.leaveThread()
+                }
+            }
+        }
+        messageListView.setMessageReplyHandler { _, message ->
+            messageComposerViewModel.performMessageAction(Reply(message))
+        }
+        messageListView.setMessageEditHandler { message ->
+            messageComposerViewModel.performMessageAction(Edit(message))
+        }
+    }
+}
+```
+
+In the snippet above, you initialize the message composer and integrate it with the `MessageListView` by passing actions from the message list to the composer.
+
```

Comment:
Before the next heading can we add an image of what the chat fragment provides as UI? I know it's a bit dumb because we have docs on it and the sample, but it will serve as a way to show how it's no different from the InputView